### ISSUE TITLE
InfluxDB Error Response: missing required parameter "q"

### ISSUE THREAD

**Issue by cloud-rocket** [grafana/grafana#3042]
Trying to query data with Grafana from InfluxDB.

When passing the query to InfluxDB admin, it is working fine.

Grafana detects tags and possible values.

InfluxDB 0.9.4.2
Grafana v2.1.3

Thanks

**Comment by torkelo** [comment]
Have you specified a measurement name? Check chrome dev tools network tab and inspect the query Grafana sends. 

When do you get this? Can you show the query in grafana query editor and the raw query grafana sends to InfluxDB (chrome dev tools network tab)

**Comment by cloud-rocket** [comment]
Ok, it's a little different than the first observation:

The query (decoded from Chrome) is as the following:
SELECT mean(val0) FROM "train_FD004" WHERE "unit" = '1' AND time > -7200s and time < -6820s GROUP BY time(100ms) fill(0)

But should be:
SELECT mean(val0) FROM "train_FD004" WHERE "unit" = '1' AND time > '1970-01-01 00:00:00' AND time < '1970-01-01 00:10:00'  GROUP BY time(100ms) fill(0)

(See attached photo)

BTW: when you click on "eye" to the left of  "select" query the server receives an empty query, causing the "missing parameter" error:
https://bone.predict.imubit.com:8086/query?db=turbofan_simulation&epoch=ms&p=xBFKVXLENoRRLW1iTxTY&q=&u=root

![grafana_query](https://cloud.githubusercontent.com/assets/2440892/10739541/b5c4898c-7c25-11e5-8a8c-7b4edfa4576d.PNG)

**Comment by torkelo** [comment]
you need to select a valid time range using the time picker. Also the eye disables the query, so when all are disabled no query is sent

**Comment by cloud-rocket** [comment]
Why this time range is not valid? (the time series stored in InfluxDB is not based on real dates, but contains 1s samples), this is why it appears as 1970-01-01..... Grafana does not support dates in past?

An eye disables the query, but triggers empty request.....

**Comment by torkelo** [comment]
not real dates?  sure it supports real dates, try picking "Last 1 hour" or any time range after 1970. 

The query sent to InfluxDB looks correct (it sends unix epocs in seconds). But your time range is invalid it needs to be before Thu Jan 01 1970 01:00:00 GMT+0100 (CET) to be a positive unix epoch

**Comment by cloud-rocket** [comment]
Owh! Sorry....
The problem was I was not on GMT....

**Comment by sveip** [comment]
@torkelo I get this error even I have chosen a time (10m) in the graphical interface. If I switch editor mode the query has ` AND $timeFilter ` specified, so it seems the value is not properly set? 

Any idea what it might be?

**Comment by Dual-Boot** [comment]
Hello,

 I have this problem since today after a whole upgrade, so I checked my local dpkg history (I use Fabric to upgrade all my servers) and I can see this : 

```
2016-05-02 21:29:41 upgrade grafana:amd64 3.0.0-beta51460725904 3.0.0-beta71462173753
2016-05-02 21:29:41 status half-configured grafana:amd64 3.0.0-beta51460725904
2016-05-02 21:29:41 status unpacked grafana:amd64 3.0.0-beta51460725904
2016-05-02 21:29:41 status half-installed grafana:amd64 3.0.0-beta51460725904
2016-05-02 21:29:43 status half-installed grafana:amd64 3.0.0-beta51460725904
2016-05-02 21:30:17 status half-installed grafana:amd64 3.0.0-beta51460725904
2016-05-02 21:30:18 status unpacked grafana:amd64 3.0.0-beta71462173753
2016-05-02 21:30:23 status unpacked grafana:amd64 3.0.0-beta71462173753
2016-05-02 21:30:25 configure grafana:amd64 3.0.0-beta71462173753 <aucun>
2016-05-02 21:30:25 status unpacked grafana:amd64 3.0.0-beta71462173753
2016-05-02 21:30:26 status unpacked grafana:amd64 3.0.0-beta71462173753
2016-05-02 21:30:26 status unpacked grafana:amd64 3.0.0-beta71462173753
2016-05-02 21:30:26 status unpacked grafana:amd64 3.0.0-beta71462173753
2016-05-02 21:30:26 status unpacked grafana:amd64 3.0.0-beta71462173753
2016-05-02 21:30:26 status unpacked grafana:amd64 3.0.0-beta71462173753
2016-05-02 21:30:26 status half-configured grafana:amd64 3.0.0-beta71462173753
2016-05-02 21:30:28 status installed grafana:amd64 3.0.0-beta71462173753
```

 At fisrt look it seems to be related to Grafana update.

The problem appears with Nginx/Apache status. Here a sample : 
```
SELECT non_negative_derivative("accepts", 10s) FROM "nginx" WHERE "host" = 'srv-dev01' AND "server" = 'localhost' AND "port" = 82 AND $timeFilter
```
And the same sample using the graphical view/query : 
![selection_048](https://cloud.githubusercontent.com/assets/5793775/14967974/7e58510c-10ba-11e6-8897-898d50a738ac.png)

The server is running Ubuntu 14.04 64 Bit
Linux srv-influxdb01 3.16.0-70-generic #90~14.04.1-Ubuntu SMP Wed Apr 6 22:56:34 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
grafana-server -v
Version 3.0.0-beta7 (commit: v3.0.0-beta7)

Regards,

**Comment by sveip** [comment]
@torkelo @Dual-Boot As a note, I also tried the beta 3.0 beta first with the same problem, later to revert to latest stable as of yesterday, still experiencing the same problem. So there might be a problem with the first installation that "contaminated" the new install.

**Comment by Dual-Boot** [comment]
you should use `aptitude purge grafana` before reinstalling stable version.
it's more cleaner.
And try to reinstall the stable version.

**Comment by sveip** [comment]
I did uninstall from software center, so I guess that is what was done.

**Comment by Dual-Boot** [comment]
Nope. Software center does not made purge.

**Comment by andrecarlucci** [comment]
Having the same problem here, did you manage to figure it out?

**Comment by andrecarlucci** [comment]
Well, it worked. I solved manually writing the query :)

**Comment by Dual-Boot** [comment]
Hi,

 it will be nice to show a sample of the working query.

 Thank you.

**Comment by andrecarlucci** [comment]
SELECT sum("value") FROM "EneatDel" WHERE "Medidor" = '2254' and time >= '2015-01-01' AND time <= '2015-01-10'  GROUP BY time(1h)

That was the one.

**Comment by sveip** [comment]
What was it when you got the error?

-----Original Message-----

**Comment by ship1999** [comment]
Now, I get this error. My version of influxdb is 0.13.0, and version of grafana is 3.0. :(

**Comment by andrecarlucci** [comment]
When using Grafana for Windows I don't get the error and I can use the
interface without any problems :/

On Thu, 19 May 2016 at 01:38 ship1999 <notifications@github.com> wrote:

> Now, I get this error. My version of influxdb is 0.13.0, and version of
> grafana is 3.0. :(
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/grafana/grafana/issues/3042#issuecomment-220225765>
>
-- 
André Carlucci
twitter/andrecarlucci
Microsoft MVP
Intel Innovator, Black Belt
www.andrecarlucci.com

**Comment by dky** [comment]
@ship1999, did you get to the bottom of this? I just installed 0.13.0 and 3.0 and getting the same error.

**Comment by CJCShadowsan** [comment]
@ship1999 - I'll add a "me too" to this. influxdb 0.13.0 and grafana 3.04 - same problem. I'm met with a missing required parameter error.

**Comment by torkelo** [comment]
@CJCShadowsan can you show contents of your chrome dev tools network tab, the raw request send by Grafana to InfluxDB and the raw response?

**Comment by CJCShadowsan** [comment]
GET **/api/datasources/proxy/5/query?db=collectd&q=&epoch=ms HTTP/1.1**
Host: 10.28.9.154:3000
Connection: keep-alive
Accept: application/json, text/plain, */*
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.94 Safari/537.36
DNT: 1
Referer: http://10.28.9.154:3000/dashboard/db/test
Accept-Encoding: gzip, deflate, sdch
Accept-Language: en-US,en;q=0.8
Cookie: grafana_sess=79ebe6437348e60a; grafana_user=admin; grafana_remember=503943298baa8f0031848e8d84a3f1495877012000730afb

Think that'll do it... &q=<blank>&epoch=ms ?

**Comment by torkelo** [comment]
there is no query there, have you specified a query in Grafana? What does your metrics tab look like for this graph?

**Comment by CJCShadowsan** [comment]
@torkelo  - more information:

Panel data error: Object
GET http://10.28.9.154:3000/api/datasources/proxy/5/query?db=collectd&q=&epoch=ms 400 (Bad Request)

Object {data: Object, status: 400, config: Object, statusText: "Bad Request"}

Object {message: "InfluxDB Error Response: missing required parameter "q"", data: Object, config: Object}config: Objectdata: nullheaders: Objectinspect: Objectmethod: "GET"paramSerializer: (a)params: Objectprecision: "ms"retry: 0transformRequest: Array[1]transformResponse: Array[1]url: "/api/datasources/proxy/5/query"__proto__: Objectdata: Objectmessage: "InfluxDB Error Response: missing required parameter "q""__proto__: Object

Panel data error: Object {message: "InfluxDB Error Response: missing required parameter "q"", data: Object, config: Object}config: Objectdata: nullheaders: Objectinspect: Objectmethod: "GET"paramSerializer: (a)params: Objectdb: "collectd"epoch: "ms"p: undefinedq: ""u: undefined__proto__: Objectprecision: "ms"retry: 0transformRequest: Array[1]transformResponse: Array[1]url: "/api/datasources/proxy/5/query"__proto__: Objectdata: Objecterror: "missing required parameter "q""message: "missing required parameter "q""__proto__: Objectmessage: "InfluxDB Error Response: missing required parameter "q""__proto__: Object__defineGetter__: __defineGetter__()__defineSetter__: __defineSetter__()__lookupGetter__: __lookupGetter__()__lookupSetter__: __lookupSetter__()constructor: Object()hasOwnProperty: hasOwnProperty()isPrototypeOf: isPrototypeOf()propertyIsEnumerable: propertyIsEnumerable()toLocaleString: toLocaleString()toString: toString()valueOf: valueOf()get __proto__: get __proto__()set __proto__: set __proto__()

GET http://10.28.9.154:3000/api/datasources/proxy/5/query?db=collectd&q=&epoch=ms 400 (Bad Request)

Object {data: Object, status: 400, config: Object, statusText: "Bad Request"}

Object {message: "InfluxDB Error Response: missing required parameter "q"", data: Object, config: Object}

Panel data error: Object {message: "InfluxDB Error Response: missing required parameter "q"", data: Object, config: Object}

**Comment by torkelo** [comment]
And what does your panel metrics tab look like? what does your metric query look like there? Can you also show you dashboard json?

**Comment by CJCShadowsan** [comment]
![screen shot 2016-06-02 at 16 42 55](https://cloud.githubusercontent.com/assets/16232240/15751193/2b67c088-28e1-11e6-94a8-94eb7fd7b0d4.png)

There's the Metric query...

**Comment by CJCShadowsan** [comment]
And here's the JSON:

`{
  "id": 1,
  "title": "Test",
  "originalTitle": "Test",
  "tags": [],
  "style": "dark",
  "timezone": "browser",
  "editable": true,
  "hideControls": false,
  "sharedCrosshair": false,
  "rows": [
    {
      "collapse": false,
      "editable": true,
      "height": "250px",
      "panels": [
        {
          "aliasColors": {},
          "bars": false,
          "datasource": "collectd",
          "editable": true,
          "error": false,
          "fill": 1,
          "grid": {
            "threshold1": null,
            "threshold1Color": "rgba(216, 200, 27, 0.27)",
            "threshold2": null,
            "threshold2Color": "rgba(234, 112, 112, 0.22)"
          },
          "id": 1,
          "isNew": true,
          "legend": {
            "avg": false,
            "current": false,
            "max": false,
            "min": false,
            "show": true,
            "total": false,
            "values": false
          },
          "lines": true,
          "linewidth": 2,
          "links": [],
          "nullPointMode": "connected",
          "percentage": false,
          "pointradius": 5,
          "points": false,
          "renderer": "flot",
          "seriesOverrides": [],
          "span": 12,
          "stack": false,
          "steppedLine": false,
          "targets": [
            {
              "dsType": "influxdb",
              "groupBy": [
                {
                  "params": [
                    "auto"
                  ],
                  "type": "time"
                },
                {
                  "params": [
                    "previous"
                  ],
                  "type": "fill"
                }
              ],
              "hide": true,
              "measurement": "cpu_load_short",
              "policy": "default",
              "refId": "A",
              "resultFormat": "time_series",
              "select": [
                [
                  {
                    "params": [
                      "value"
                    ],
                    "type": "field"
                  },
                  {
                    "params": [],
                    "type": "mean"
                  }
                ]
              ],
              "tags": [],
              "query": "SELECT mean(\"value\") FROM \"cpu_load_short\" GROUP BY time($interval) fill(previous)",
              "rawQuery": true
            }
          ],
          "timeFrom": null,
          "timeShift": null,
          "title": "Woop",
          "tooltip": {
            "msResolution": false,
            "shared": true,
            "value_type": "cumulative"
          },
          "type": "graph",
          "xaxis": {
            "show": true
          },
          "yaxes": [
            {
              "format": "short",
              "label": null,
              "logBase": 1,
              "max": null,
              "min": null,
              "show": true
            },
            {
              "format": "short",
              "label": null,
              "logBase": 1,
              "max": null,
              "min": null,
              "show": true
            }
          ],
          "interval": ">1s"
        }
      ],
      "title": "Row"
    },
    {
      "collapse": false,
      "editable": true,
      "height": "250px",
      "panels": [],
      "title": "New row"
    }
  ],
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {
    "refresh_intervals": [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d"
    ],
    "time_options": [
      "5m",
      "15m",
      "1h",
      "6h",
      "12h",
      "24h",
      "2d",
      "7d",
      "30d"
    ]
  },
  "templating": {
    "list": []
  },
  "annotations": {
    "list": []
  },
  "schemaVersion": 12,
  "version": 0,
  "links": []
}`

**Comment by torkelo** [comment]
the query is disabled (you must have clicked the eye icon). Click the eye icon to the right again to re-enable it.

**Comment by CJCShadowsan** [comment]
...
Fail.

Ok so now all is working 👍

**Comment by AshishKhatkar** [comment]
@torkelo I am having issue while using Grafana. Influxdb doesn't allow me to add epoch time like 123456789s but Grafana sends query with 's' appended to the epoch time. Resulting query that Grafana sends is `SELECT "x" FROM "y" WHERE "z" = 'abc' AND time > 1451586600s and time < 1483209000s GROUP BY "w"`. I tried this query in influxdb and it doesn't returns anything. If I type this query `SELECT "x" FROM "y" WHERE "z" = 'abc' AND time > 1451586600 and time < 1483209000 GROUP BY "w"`. This works perfectly. 
I am new to Grafana and influxdb. I tried searching about this issue but couldn't find anything.

**Comment by gkpraveen1988** [comment]
Hi Team, 
Still facing the same issues. 
on the After adding the query the panel shows 

"InfluxDB Error Response: missing required parameter "q""

Tried to add a new graph -> adding the query -> hit the eye icon. Still the query is not populating. 

Failed to load resource: the server responded with a status of 400 (Bad Request)

Objectconfig: Objectdata: Objectmessage: "InfluxDB Error Response: missing required parameter "q""__proto__: Object(anonymous function) @ boot.4595d54e.js:35
boot.4595d54e.js:40 Panel data error: Objectconfig: Objectdata: nullheaders: Objectinspect: Objectmethod: "GET"paramSerializer: (a)params: Objectprecision: "ms"retry: 0transformRequest: Array[1]transformResponse: Array[1]url: "http://10.0.0.17:8086/query"__proto__: Objectdata: Objectmessage: "InfluxDB Error Response: missing required parameter "q""__proto__: Object

Can anyone please help..

**Comment by torkelo** [comment]
"hit the eye icon" you mean you hide the query, any other query? what version are you using?

**Comment by gkpraveen1988** [comment]
Grafana = 3.0.2

Initially hit the "eye" icon, then entered the query (input parameters) and again hit the "eye" icon

**Comment by torkelo** [comment]
ok, why did you hit the eye icon? did you have other queries and wanted to disable that query? 

The error is fixed in 3.1.1 (so if all queries are disabled no request to InfluxDB are sent, the error you get is due to sending a request to influxdb when all queries are disabled)
